### DISCUSSION TITLE
Links & Accounts, txt files, csv, json or all

### DISCUSSION THREAD

**Discussion by markmelnic** [markmelnic/reddit-bot#7] [Q&A]
As you might have noticed, links and accounts files are passed as path arguments for the bot to do what it has to. After reviewing the code a bit I realized the current approach is by far not the best, so I figured why not change the way we store and process these files.

The alternatives I came up with are:
- CSV: Structured data, no need for separators in the txt file, has its downsides (due to the comma separation it becomes a forbidden character, but ofc there are workarounds to this, such as `.tsv` files) but generally a good format to work with
- JSON: Probably what I would prefer. Store all links and accounts in a single file, but requires some manual formatting on the side of the user and is more problematic for users who haven't worked with or don't understand this file format.
- all file formats: Pass in any file format and let the bot figure out how to parse the data, more complex on the development side but offers the best of all worlds.

Please let me know your thoughts and what you consider to be the best approach. If you have any ideas or suggestions I have not mentioned here, feel free to let me know. Until that, thanks!

**Comment by britcanamedia** [comment]
I store the links and content sample (comment/post) in the DB (max 3000 chars), the accounts in a secondary table (username,password,ip_address,failed_login,platform [reddit/instagram/tiktok/youtube]) I then have those accounts assigned to specific IPs that only that account can ever use for its life time (/16). Then I utilize sockets and threading on a MASSIVE amazon instance (2000 gigs of ram) to connect outbound on those IPs and do its thing. It can handle about 100 Chromium instances at the same time, no problem.

**Reply by markmelnic** [reply]
That is That is definitely an interesting use case. My performance considerations are of a user who is running the bot on their local machine, this is why even a couple Chromium instances can start throttling. Connecting to a database or passing the users, content etc. from a database session would be an interesting addition to the bot, I'll add that to my "someday" list )

**Reply by britcanamedia** [reply]
I'd be happy to contribute
